src/common/SharedMemory.cpp

```diff
@@ -23,45 +23,44 @@
 
 #include "SharedMemory.h"
 
+#include <system_error>
+#include <utility>
+
 #include "lmmsconfig.h"
+#include "RaiiHelpers.h"
 
 #ifdef LMMS_HAVE_UNISTD_H
 #	include <unistd.h>
 #endif
 
-#if _POSIX_SHARED_MEMORY_OBJECTS > 0
-#	include <system_error>
-#
+#if _POSIX_SHARED_MEMORY_OBJECTS > 0 || defined(LMMS_BUILD_APPLE)
 #	include <sys/mman.h>
 #	include <sys/stat.h>
 #	include <fcntl.h>
-#
-#	include "RaiiHelpers.h"
+#elif defined(LMMS_BUILD_WIN32)
+#	include <windows.h>
 #else
-#	include <stdexcept>
-#
-#	include <QtGlobal>
-#	include <QSharedMemory>
+#	error "No shared memory implementation available"
```

Comment:
Is there any chance this might break compilation for some lesser used OS es like BSD?

Reply:
There is, yes. The current code will work on modern POSIX systems, as well as Windows. BSD variants should be fine, but there may be other systems where this doesn't work. I'd rather patch in support for those as we come across them, however, as otherwise we need to maintain the Qt dependency in CMake code.